lib/DateTimeParser.php

```diff
@@ -86,80 +81,90 @@ public static function parseDate($date, DateTimeZone $tz = null)
     /**
      * Parses an iCalendar (RFC5545) formatted duration value.
      *
-     * This method will either return a DateTimeInterval object, or a string
-     * suitable for strtotime or DateTime::modify.
-     *
-     * @param string $duration
-     * @param bool   $asString
+     * This method will return a DateTimeInterval object
      *
-     * @return DateInterval|string
+     * @throws InvalidDataException
+     * @throws Exception
      */
-    public static function parseDuration($duration, $asString = false)
+    public static function parseDuration(string $duration): DateInterval
```

Comment:
ToDo: understand why we can remove the `$asString` parameter here. This parameter is never used internally, but people using `sabre/vobject` could call this `Sabre\VObject\DateTimeParser::parseDuration` static function and us the `$asString` parameter.

@tcitworld do you remember why you removed the parameter?

Reply:
I probably thought it would be nicer to have consistent parameters types and return types. Sounds a sane BC.

Reply:
I will start making notes for the release notes - need to mention this sort of stuff.

Reply:
we can do such changes more easily/with confidence when we declare the classes which are not designed to be subclasses as `final` that way.